lib/resources/package.rb

```diff
@@ -44,8 +44,10 @@ def initialize(package_name = nil) # rubocop:disable Metrics/AbcSize
       elsif ['hpux'].include?(os[:family])
         @pkgman = HpuxPkg.new(inspec)
       else
-        return skip_resource 'The `package` resource is not supported on your OS yet.'
+        skip_resource 'The `package` resource is not supported on your OS yet.'
       end
+
+      evaluate_missing_requirements(@pkgman.missing_requirements) if @pkgman
```

Comment:
If you leave the `return` above, you'll always have a populated `@pkgman` variable, so I'd just remove `if @pkgman`